src/main/java/io/horrorshow/codey/time/DiscordTimezone.java

```diff
@@ -0,0 +1,135 @@
+package io.horrorshow.codey.time;
+
+import io.horrorshow.codey.discordutil.DiscordUtils;
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.util.Arrays;
+
+@Service
+public class DiscordTimezone extends ListenerAdapter {
+    private final DiscordUtils utils;
+
+    public DiscordTimezone(@Autowired JDA jda, @Autowired DiscordUtils utils) {
+        this.utils = utils;
+
+        jda.addEventListener(this);
+    }
+
+    private final String emoji = "⏰";
+    private final String deleteEmoji = "\uD83D\uDDD1";
+
+    @Override
+    public void onGuildMessageReceived(@NotNull GuildMessageReceivedEvent event) {
+        if (!event.getAuthor().isBot()) {
+            onMessage(event.getMessage());
+        }
+    }
+
+    private void onMessage(Message message) {
+        String rawContent = message.getContentRaw();
+        String[] rawContentParts = rawContent.split("\\s+");
+
+        if(rawContentParts.length == 0)
+            return;
+
+        Object[] result = time(rawContentParts);
+
+        if(!((boolean) result[0])) {
```

Comment:
This is unreadable: `!((boolean) result[0])`, from just looking at it without having studied the rest of the code it's very cryptic, same with the statement below. But this can be cleaned up after the time() method has been restructured